exchanges/okx/okx_websocket.go

```diff
@@ -672,27 +738,156 @@ func (ok *Okx) WsHandleData(respRaw []byte) error {
+			Data      []WsWithdrawlInfo `json:"data"`
+		}{}
+		return ok.wsProcessPushData(respRaw, resp)
+	case okxRecurringBuyChannel:
+		resp := &struct {
+			Arguments SubscriptionInfo    `json:"arg"`
+			Data      []RecurringBuyOrder `json:"data"`
+		}{}
+		return ok.wsProcessPushData(respRaw, resp)
+	case okxLiquidationOrders:
+		var resp LiquidiationOrder
+		return ok.wsProcessPushData(respRaw, &resp)
+	case okxADLWarning:
+		var resp ADLWarning
+		return ok.wsProcessPushData(respRaw, &resp)
+	case okxEconomicCalendar:
+		var resp EconomicCalendarResponse
+		return ok.wsProcessPushData(respRaw, &resp)
+	case okxCopyTrading:
+		var resp CopyTradingNotification
+		return ok.wsProcessPushData(respRaw, &resp)
 	default:
 		ok.Websocket.DataHandler <- stream.UnhandledMessageWarning{Message: ok.Name + stream.UnhandledMessage + string(respRaw)}
 		return nil
 	}
 }
 
+// wsProcessSpreadTrades handle and process spread order trades
+func (ok *Okx) wsProcessSpreadTrades(respRaw []byte) error {
+	if respRaw == nil {
+		return common.ErrNilPointer
+	}
+	var resp WsSpreadOrderTrade
+	err := json.Unmarshal(respRaw, &resp)
+	if err != nil {
+		return err
+	}
+	if len(resp.Data) == 0 {
+		return kline.ErrNoTimeSeriesDataToConvert
+	}
+	pair, err := ok.GetPairFromInstrumentID(resp.Argument.SpreadID)
+	if err != nil {
+		return err
+	}
+	trades := make([]trade.Data, len(resp.Data))
+	for x := range resp.Data {
+		oSide, err := order.StringToOrderSide(resp.Data[x].Side)
+		if err != nil {
+			return err
+		}
+		trades[x] = trade.Data{
+			Amount:       resp.Data[x].FillSize.Float64(),
+			AssetType:    asset.Spread,
+			CurrencyPair: pair,
+			Exchange:     ok.Name,
+			Side:         oSide,
+			Timestamp:    resp.Data[x].Timestamp.Time(),
+			TID:          resp.Data[x].TradeID,
+			Price:        resp.Data[x].FillPrice.Float64(),
+		}
+	}
+	return trade.AddTradesToBuffer(ok.Name, trades...)
+}
+
+// wsProcessSpreadOrders retrieve order information from the sprd-order Websocket channel.
+// Data will not be pushed when first subscribed.
+// Data will only be pushed when triggered by events such as placing/canceling order.
+func (ok *Okx) wsProcessSpreadOrders(respRaw []byte) error {
+	if respRaw == nil {
+		return common.ErrNilPointer
+	}
+	resp := &struct {
+		Argument SubscriptionInfo `json:"arg"`
+		Data     []WsSpreadOrder  `json:"data"`
+	}{}
+	err := json.Unmarshal(respRaw, &resp)
+	if err != nil {
+		return err
+	}
+	if len(resp.Data) == 0 {
+		return kline.ErrNoTimeSeriesDataToConvert
+	}
+	pair, err := ok.GetPairFromInstrumentID(resp.Argument.InstrumentID)
+	if err != nil {
+		return err
+	}
+
+	for x := range resp.Data {
+		oSide, err := order.StringToOrderSide(resp.Data[x].Side)
+		if err != nil {
+			return err
+		}
+		oStatus, err := order.StringToOrderStatus(resp.Data[x].State)
+		if err != nil {
+			return err
+		}
+		oType, err := order.StringToOrderType(resp.Data[x].OrderType)
+		if err != nil {
+			return err
+		}
+		d := &order.Detail{
```

Comment:
If you can append to a slice and then send it over the channel as a slice. And could you also add in slice handling in engine?

Reply:
I have changed the data to be handed to the channel to a slice. But we already have a slice of order data handling in the **_websocketroutine_manager.go_** 
<img width="703" alt="Screenshot 2024-10-29 at 9 51 39 AM" src="https://github.com/user-attachments/assets/e3d657a3-4738-4580-9d75-193a03067f66">